internal/utils/utils.go

```diff
@@ -288,3 +288,53 @@ func Exit(exitCode int, err string) {
 func FlattenPath(s string) string {
 	return strings.Replace(s, "/", "__", -1)
 }
+
+// Expand function is similar to os.Expand, with two differences:
+// * "$$" is always converted to "$"
```

Comment:
If we're leaving things unchanged if we don't recognize the expansion, then shouldn't we drop this $$ behaviour.
We only need it if we always replaced $anything, to stop the expansion.